CONTRIBUTING.md

```diff
@@ -68,38 +69,97 @@ Please see `./ci/features`.
 To test the usage of the Conjur Service Broker within a CF deployment, you can
 follow the demo scripts in the [Cloud Foundry demo repo](https://github.com/conjurinc/cloudfoundry-conjur-demo).
 
-To run the test suite on your local machine:
+#### Running the Dev Environment
+To test your changes within a running instance of Cloud Foundry and Conjur, run:
+
+```shell script
+./ci/start_dev_environment
+```
+
+This starts Conjur and Cloud Foundry containers, and provides terminal 
+access to the Cloud Foundry container. You do not need to restart the container
+after you make changes to the project.
+
+To run the `cucumber` within the development environment, run the following command from
+the `ci` directory, within the container:
+
+```shell script
+cucumber \
+    --format pretty \
+    --format junit \
+    --out ./features/reports \
+    --tags 'not @integration'
 ```
-$ ./package.sh   # Create the conjur-env binary in the vendor dir and a ZIP of the project contents
-$ ./test.sh      # Run the test suite
+
+#### Setup
+To run the test suite on your local machine, first run:
+```shell script
+./package.sh
 ```
+This will create the conjur-env binary in the vendor dir and a ZIP of the project contents.
 
 #### Integration Testing
 
 To run the buildpack integration tests, the test script needs to be given the API endpoint and admin credentials
 for a CloudFoundry installation. These are provided as environment variables to the script:
 
-```sh-session
-$ export CF_API_ENDPOINT=https://api.sys.cloudfoundry.net
-$ CF_ADMIN_PASSWORD=... ./test.sh
+```shell script
+export CF_API_ENDPOINT=https://api.sys.cloudfoundry.net
+CF_ADMIN_PASSWORD=... ./ci/test_integration.sh
 ```
 
 These variables may also be provided using [Summon](https://cyberark.github.io/summon/) by updating the `secrets.yml`
 file as needed and running:
 
+```shell script
+summon ./ci/test_e2e
```

Comment:
Should the `secrets.yml` be moved to be adjacent (same directory) as `ci/test_e2e`? I think that's the only script that will need the secrets.yml.